### PR TITLE
Always initialize the collections

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/Azure/autorest.csharp/issues/521

I've split the PR into multiple commits:

First one is the most important to review, second is resulting generated code change and the third is some follow up changes.

### CODE REVIEW

File: test/AutoRest.Shared.Tests/ChangeTrackingListTest.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+            list.CopyTo(new string[5], 0);
+
+            Assert.Throws<IndexOutOfRangeException>(() => _ = list[0]);
+
+            Assert.True(list.IsUndefined);
+        }
+
+        [Test]
+        public void CanAddElement()
```
Comment (ShivangiReja) [inline test/AutoRest.Shared.Tests/ChangeTrackingListTest.cs:40]: Tests are missing for some operations like GetEnumerator(), Insert(), contains()...etc

File: src/AutoRest.CSharp.V3/Output/Builders/SerializationBuilder.cs [hunk 2]
```diff
@@ -179,7 +178,7 @@ private IEnumerable<JsonPropertySerialization> GetPropertySerializationsFromBag(
                 JsonPropertySerialization[] serializationProperties = GetPropertySerializationsFromBag(innerBag, objectType).ToArray();
                 JsonObjectSerialization objectSerialization = new JsonObjectSerialization(null, serializationProperties, null);
                 yield return new JsonPropertySerialization(
-                    name, false,null, objectSerialization);
+                    name,null, objectSerialization);
```
Comment (ShivangiReja) [inline src/AutoRest.CSharp.V3/Output/Builders/SerializationBuilder.cs:181]: nit: indentation is off

File: src/assets/Generator.Shared/ChangeTrackingExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+using System.Linq;
+
+#nullable disable
+
+namespace Azure.Core
+{
+    internal static class ChangeTrackingExtensions
+    {
+    }
+}
```
Comment (ShivangiReja) [inline src/assets/Generator.Shared/ChangeTrackingExtensions.cs:16]: Why do we have this empty class? I don't see we are using it anywhere.

File: test/AutoRest.Shared.Tests/ChangeTrackingListTest.cs (hunk 1 above)
Comment (pakrym) [reply test/AutoRest.Shared.Tests/ChangeTrackingListTest.cs:40]: GetEnumerator is covered by foreach. You are right about the others. Adding.

**Pull request merged**